src/arnoldi.jl

```diff
@@ -101,9 +101,10 @@ function arnoldi(A, b; m=min(30, size(A, 1)), ishermitian=LinearAlgebra.ishermit
     n = length(b)
     U = ishermitian ? real(T) : T
 
-    V = similar(A, T, (n, m + 1))
-    H = similar(A, U, (m+1, m))
-    fill!(H, zero(U))
+    # V stores the Krylov vectors
+    V = similar(b, T, (n, m + 1))
+    # H is a small dense array in tridiagonal form
+    H = zeros(U, (m+1, m))
```

Comment:
`H` will alway be `Matrix` if you don't use `similar`. Is that intended?

Reply:
Yes, it is intended. This matrix is usually a very small one that can be easily diagonalized on CPU.
Also, there are a lot of element-wise operations to it.